portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,114 @@
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.classes.Comment;
 import java.io.IOException;
+import static java.lang.Math.toIntExact;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comment")
 public class DataServlet extends HttpServlet {
+  private final DatastoreService dataStore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.setContentType("application/json");
+    Query query = new Query("Comment").addSort("likes", SortDirection.DESCENDING);
+
+    PreparedQuery comments = dataStore.prepare(query);
+    List<Comment> commentsToSet = new ArrayList<>();
+    for (Entity entity : comments.asIterable()) {
+        String user = (String) entity.getProperty("user");
+        String commentText = (String) entity.getProperty("comment");
+        long commentSize = commentText.length();
+        long time = (long) entity.getProperty("time");
+        Long likes = (Long) entity.getProperty("likes");
+        
+        long id = entity.getKey().getId();
+
+        Comment commentToSet = new Comment(commentText, user, commentSize, likes.intValue(), time, id);
+        commentsToSet.add(commentToSet);
+    }
+
+    Gson gson = new Gson();
+    String payload = gson.toJson(commentsToSet);
+
+    response.getWriter().println(payload);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    String commentText = request.getParameter("comment");
+    String user = request.getParameter("user");
+    long commentSize = commentText.length();
+    long time = System.currentTimeMillis();
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("user", user);
+    commentEntity.setProperty("comment", commentText);
+    commentEntity.setProperty("time", time);
+    commentEntity.setProperty("size", commentSize);
+    commentEntity.setProperty("likes", 0);
+
+    dataStore.put(commentEntity);
+
+    long id = commentEntity.getKey().getId();
+
+    Comment comment = new Comment(commentText, user, commentSize, 0, time, id);
```

Comment:
It looks like all the field on your Comment class and your DataStore entity are pretty much the same, which makes sense.

Might be worth adding a ToEntity() function on your Comment class and moving the Entity creation logic there. That way you can also enforce the `id` field is set after the Entity is created.